Add release notes for v0.1.9 (#1139)
VirtusLab/scala-cli: .github/release/release_notes.md

# [v0.1.9](https://github.com/VirtusLab/scala-cli/releases/tag/v0.1.9)

## Force interactive mode for `update` command

Interactive mode for `update` sub-command is now enabled by default. 

Added in https://github.com/VirtusLab/scala-cli/pull/1100 by @lwronski

## `--list-main-classes` for `publish` & `package`

> `publish` and `package` sub-commands handle `--list-main-classes` option, which lists all the available main classes.

Reviewer: ```suggestion
`publish` and `package` sub-commands now support the `--list-main-classes` option, which allows to list all the available main classes. Previously it was only available in the `run` command.
```

> Added in https://github.com/VirtusLab/scala-cli/pull/1118  by @Gedochao

Reviewer: ```suggestion
Added in https://github.com/VirtusLab/scala-cli/pull/1118 by @Gedochao
```
NIT: unnecessary extra space

> ## Fixes

Reviewer: ```suggestion
## Important fixes & enhancements
```

### `fmt` options improvement

> It is now possible to pass native `scalafmt` options in `fmt` sub-command with the `-F`. 

Reviewer: Technically speaking, it was possible for a while now, but after this change we added the relevant documentation and made the option publicly visible... So maybe something like:
```suggestion
Added missing documentation on how to pass native `scalafmt` options in the `fmt` sub-command with the `-F` option.
```

```
$ scala-cli fmt -F --version
scalafmt 3.5.2
```

> 

Reviewer: ```suggestion

Additionally, a couple of `scalafmt`'s native options received aliases in Scala CLI: 

```

> `--respect-project-filters` is turned on by default in Scala CLI. This addresses https://github.com/VirtusLab/scala-cli/issues/1121

Reviewer: ```suggestion
`--respect-project-filters` is an alias for `-F --respect-project-filters`. Because of the way sources are passed by Scala CLI to `scalafmt` under the hood, we now turn it on by default to respect any `project.excludePaths` settings in the user's `.scalafmt.conf`.  
It can be disabled by passing `--respect-project-filters=false` to revert to previous behaviour.

This addresses https://github.com/VirtusLab/scala-cli/issues/1121
```

> 

Reviewer: ```suggestion

`--scalafmt-help` is an alias for `-F --help`. It shows the `--help` output from `scalafmt`, which might prove as helpful reference when in need of using native `scalafmt` options with `-F`.

```

Added in https://github.com/VirtusLab/scala-cli/pull/1135 by @Gedochao

> ### libsodium.dll is not missing on windows

Reviewer: ```suggestion
### Include `libsodium.dll` on windows
```

Added in https://github.com/VirtusLab/scala-cli/pull/1118 by @Gedochao

## Important fixes & enhancements

### `fmt` options improvement

Added missing documentation on how to pass native `scalafmt` options in the `fmt` sub-command with the `-F` option.
```
$ scala-cli fmt -F --version
scalafmt 3.5.2
```

Additionally, a couple of `scalafmt`'s native options received aliases in Scala CLI: 

`--respect-project-filters` is an alias for `-F --respect-project-filters`. Because of the way sources are passed by Scala CLI to `scalafmt` under the hood, we now turn it on by default to respect any `project.excludePaths` settings in the user's `.scalafmt.conf`.  
It can be disabled by passing `--respect-project-filters=false` to revert to previous behaviour. 
This addresses https://github.com/VirtusLab/scala-cli/issues/1121

`--scalafmt-help` is an alias for `-F --help`. It shows the `--help` output from `scalafmt`, which might prove as helpful reference when in need of using native `scalafmt` options with `-F`.

Added in https://github.com/VirtusLab/scala-cli/pull/1135 by @Gedochao

> ### Include `libsodium.dll` on windows

Reviewer: ```suggestion
### Include `libsodium.dll` on Windows
```

Static linking of libsodium in Windows launcher has been fixed. 
This addresses https://github.com/VirtusLab/scala-cli/issues/1114

Added in https://github.com/VirtusLab/scala-cli/pull/1115 by @alexarchambault

### Force interactive mode for `update` command

Interactive mode for `update` sub-command is now enabled by default. 

Added in https://github.com/VirtusLab/scala-cli/pull/1100 by @lwronski

## In progress

### Publishing-related features 

* Publish tweaks + documentation by @alexarchambault in  https://github.com/VirtusLab/scala-cli/pull/1107

> ### BSP support for scala-sc

Reviewer 2: I would have phrased it this way:
```suggestion
### Better BSP support for Scala scripts
```
